internal/sqlutil/sql.go

```diff
@@ -107,3 +111,40 @@ func SQLiteDriverName() string {
 	}
 	return "sqlite3"
 }
+
+func minOfInts(a, b int) int {
+	if a <= b {
+		return a
+	}
+	return b
+}
+
+type QueryProvider interface {
+	QueryContext(ctx context.Context, query string, args ...interface{}) (*sql.Rows, error)
+}
+
+const SQLite3MaxVariables = 999
+
+//RunLimitedVariablesQuery split up a query with more varaibles than the used databse can handle in multiple queries.
+func RunLimitedVariablesQuery(ctx context.Context, query string, qp QueryProvider, rowHandler func(*sql.Rows) error, variables []interface{}) error {
+	var start int
+	for start < len(variables) {
+		n := minOfInts(len(variables)-start, SQLite3MaxVariables)
+		query := strings.Replace(query, "($1)", QueryVariadic(n), 1)
+		rows, err := qp.QueryContext(ctx, query, variables[start:start+n]...)
```

Comment:
Given we have a nice interface for this, please can you add unit tests to check this function does what you think it should be doing and we don't have off-by-one errors somewhere.

Reply:
Will do.